### ISSUE TITLE
WithImage/AddContainer: add registry argument

### ISSUE THREAD

**Issue by tmds** [dotnet/aspire#3911]
This is for adding a registry argument to `WithImage`:
```cs
WithImage<T>(this IResourceBuilder<T> builder, string image, string tag = "latest", string? registry = null)
```
which can then be used instead of: `WithImage(..).WithImageRegistry(...)`.

**Comment by timheuer** [comment]
Agreed, propsoed this as well when we discussed default registry options: https://github.com/dotnet/aspire/issues/3423#issuecomment-2041628154

**Comment by tmds** [comment]
And also:
```
IResourceBuilder<ContainerResource> AddContainer
```

If breaking changes to the API are still allowed, perhaps `registry` should be a required argument on these methods?

**Comment by timheuer** [comment]
We are locked for API, so no more breaking changes. What we've done is model the changes/correct behavior in our own resource code, ensuring that the registry full path (in the case of docker, even the /library notation) hoping that serves as additional documentation for the 'right' way.

**Comment by davidfowl** [comment]
Closing this one as we're not going to make the breaking change at this point
